app/Http/Controllers/Install/FinalizeController.php

```diff
@@ -0,0 +1,174 @@
+        } catch (Exception $e) {
+            $messages[] = $e->getMessage();
+            $config_message = trans('install.finish.config_not_written');
+            $success = true;
+        }
+
+        // write env last only if everything else succeeded
+        if ($success) {
+            try {
+                $env = $this->writeEnvFile();
+            } catch (Exception $e) {
+                $messages[] = $e->getMessage();
+                $success = false;
+            }
+        }
+
+        if ($success) {
+            session()->flush();
+        }
+
+        return view('install.finish', $this->formatData([
+            'env' => $env,
+            'config' => $config,
+            'messages' => $messages,
+            'success' => $success,
+            'config_message' => $config_message,
+        ]));
+    }
+
+    private function writeEnvFile()
+    {
+        return EnvHelper::setEnv([
+            'NODE_ID' => uniqid(),
+            'DB_HOST' => session('db.host'),
+            'DB_PORT' => session('db.port'),
+            'DB_USERNAME' => session('db.username'),
+            'DB_PASSWORD' => session('db.password'),
+            'DB_DATABASE' => session('db.database'),
+            'DB_SOCKET' => session('db.socket'),
+        ], ['INSTALL'], base_path('.env'));
+    }
+
+    private function writeConfigFile($config_contents, $config_file)
+    {
+        if (!file_exists($config_file)) {
+            $conf = fopen($config_file, 'w');
+            if ($conf !== false) {
+                if (fwrite($conf, "<?php\n") === false) {
+                    throw new Exception("We couldn't create the config.php file, please create this manually before continuing by copying the below into a config.php in the root directory of your install (typically /opt/librenms/)");
+                }
+
+                $config_contents = stripslashes($config_contents);
+                fwrite($conf, $config_contents);
+                fclose($conf);
+                return;
+            }
+
+            throw new Exception("We couldn't create the config.php file, please create this manually before continuing by copying the below into a config.php in the root directory of your install (typically /opt/librenms/)");
+        }
+    }
+
+    public static function enabled($steps): bool
+    {
+        foreach ($steps as $step => $controller) {
+            if ($step !== 'finish' && !session("install.$step")) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public static function icon(): string
+    {
+        return 'fa-check';
+    }
+
+    private function getConfigFileContents()
+    {
+        $db = session('db');
+        $install_dir = base_path();
+
+        return <<<"EOD"
+## Have a look in defaults.inc.php for examples of settings you can set here. DO NOT EDIT defaults.inc.php!
+
+### Database config
+\$config['db_host'] = '{$db['host']}';
```

Comment:
Do we actually still need this if we set `.env`?

Reply:
Not really, but I didn't want to combine this with my PR that removes that requirement.  They are a little inter-dependent though.

Reply:
Maybe I just remove the install modifications from that PR...

Reply:
Wouldn't it be nice with a fresh installer PR? :D

Reply:
I could merge them.